### PR TITLE
Add promo redirect action to WebViewController

### PR DESCRIPTION
**Pull request by kaladin9017**
#### What's this PR do?
- Remove old podcast redirect action
- Add new promo redirect action that handles older podcast promos and new promos

#### How was this tested? How should this be reviewed?
Tested locally. 

#### What are the relevant tickets?
Relevant Github issues

#### Questions / Considerations
I used the `/*` catch-all flag after all the custom routes and excluded all assets but I'm not sure if I missed anything and this could break some part of `shinetext.com`

Need to update contentful to include utm parameters when needed.

### CODE REVIEW

File: config/routes.js [hunk 1]
```diff
@@ -156,6 +130,12 @@ var routes = {
   'post /join': 'WebActionsController.join',
   'post /save-settings': 'WebActionsController.saveSettings',
   'post /sms-invite': 'WebActionsController.smsInvite',
+
+  '/:promo': {
+    controller: 'WebViewController',
+    action: 'promoRedirect',
+    skipAssets: true,
```
Comment (jonuy) [inline config/routes.js]: What's the `skipAssets` property do?

File: api/controllers/WebViewController.js [hunk 2]
```diff
@@ -108,15 +109,46 @@ module.exports = {
...
     };
 
-    let originalName = req.url.split('/')[1];
-    let redirectName = podcastDirectories[originalName];
-    return res.redirect(
-      301,
-      `${sails.config.globals.premiumShineBaseUrl}/promo/${redirectName}`
-    );
-  },
+    Promise.coroutine(function*() {
```
Comment (jonuy) [inline api/controllers/WebViewController.js]: fwiw you should be able to use async/await on aurora. Heroku's been deploying this project with Node 10.

File: config/routes.js (hunk 1 above)
Comment (kaladin9017) [reply config/routes.js]: Skips `favicon`

**Comment by kaladin9017** [comment]
Even without the second space, I think it makes sense to add `PROD` to the contentful variable names. Will Update.

**Comment by jonuy** [comment]
Ah, I was referring more to the distinction between the Contentful space for Shine Texts, the blog, and the other web app. Either way, this is still ok to merge when it's ready.

**Comment by kaladin9017** [comment]
Ohhhh 🤦‍♂️. Both could be useful. And act as pointers for where to look on contentful for the information being fetched by the API call. Will update.

**Pull request merged**